public/app/features/dashboard-scene/pages/DashboardScenePage.test.tsx

```diff
@@ -298,6 +300,62 @@ describe('DashboardScenePage', () => {
       await waitFor(() => expect(screen.queryByText('Last 6 hours')).toBeInTheDocument());
     });
   });
+
+  describe('errors rendering', () => {
+    it('should render dashboard not found notice when dashboard... not found', async () => {
+      setupLoadDashboardMockReject({
+        status: 404,
+        statusText: 'Not Found',
+        data: {
+          message: 'Dashboard not found',
+        },
+        config: {
+          method: 'GET',
+          url: 'api/dashboards/uid/adfjq9edwm0hsdsa',
+          retry: 0,
+          headers: {
+            'X-Grafana-Org-Id': 1,
+          },
+          hideFromInspector: true,
+        },
+        isHandled: true,
+      });
+
+      setup();
+
+      expect(await screen.findByTestId(selectors.components.EntityNotFound.container)).toBeInTheDocument();
+    });
+    it('should render error alert for backend errors', async () => {
+      setupLoadDashboardMockReject({
+        status: 500,
+        statusText: 'internal server error',
+        data: {
+          message: 'Internal server error',
+        },
+        config: {
+          method: 'GET',
+          url: 'api/dashboards/uid/adfjq9edwm0hsdsa',
+          retry: 0,
+          headers: {
+            'X-Grafana-Org-Id': 1,
+          },
+          hideFromInspector: true,
+        },
+        isHandled: true,
+      });
+
+      setup();
+
+      expect(await screen.findByTestId('dashboard-page-error')).toBeInTheDocument();
+    });
+    it('should render error alert for runtime errors', async () => {
+      setupLoadDashboardRuntimeErrorMock();
+
+      setup();
+
+      expect(await screen.findByTestId('dashboard-page-error')).toBeInTheDocument();
```

Comment:
It is a small nit, but we could also check the error message is present to ensure the message is not lost or that the triggered error for this test is a backend error instead of a runtime. This applies also to [backend error test case](https://github.com/grafana/grafana/pull/98950/files#diff-c4dcbead210d49072b09d1269a1553d2d3d462c0b29f56e9ef34e91df0060da3R328)

Reply:
sure, why not :) Added in f09c8b5641c267e706a5c79274f6eec9b6749cdc